### ISSUE TITLE
run coverage in github actions

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2321]
See: #2315.

**Comment by pq** [comment]
Setup a little differently than ours: https://medium.com/flutter-community/testing-dart-packages-with-github-actions-4c2c671b1e34 but maybe worth cribbing.

(NOTE: `COVERALLS_TOKEN` is set in secrets.)

**Comment by pq** [comment]
@athomas, looking over at https://github.com/dart-lang/dartdoc/pull/2464, I see you helped migrate dartdoc but there were maybe some caveats... Could you advise here?  Or would you be willing to help?

**Comment by athomas** [comment]
Sure, shouldn't be too hard. Take a look at the [pkg/coverage workflow](https://github.com/dart-lang/coverage/blob/484022221f5589fa244a396bc8e424ac1bea6233/.github/workflows/test-package.yml#L74).

The Actions part is fairly simple. I've tightened the branch protection settings so that you may use the `GITHUB_TOKEN`-style coveralls integration used elsewhere. You will not need the `COVERALLS_TOKEN` anymore.

You will also have to remove this from `travis.sh` because it will stop working when we're off Travis completely:
```
    echo "Running coveralls..."
    gem install coveralls-lcov # <-- this gem is Travis specific (will work now but break later)
    coveralls-lcov var/lcov.info
```

**Comment by pq** [comment]
Thanks @athomas!  I started playing around and hit a quick snag.  I bet  it's  an easy fix? #2432...
